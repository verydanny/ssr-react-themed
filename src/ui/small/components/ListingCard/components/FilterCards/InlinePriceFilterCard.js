import React, { PureComponent } from 'react'
import { PriceFilterCard } from '@rentpath/react-ui-core'
import autobind from 'autobind-decorator'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import { themed } from 'react-themed'
import { PRICE_FILTER_SLIDER_DEFAULTS } from './const'

@themed(/^Inline/)
export default class InlinePriceFilterCard extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    submitUrl: PropTypes.string,
    updateFilterCriteriaWithChanges: PropTypes.func.isRequired,
    pushState: PropTypes.func.isRequired,
  }

  static defaultProps = {
    theme: {},
  }

  constructor(props) {
    super(props)
    this.state = { isSliding: false }
  }

  @autobind
  handleSubmit() {
    const { pushState, submitUrl } = this.props
    pushState(submitUrl)
  }

  @autobind
  handlePriceChange(value) {
    if (window.eventTracker) {
      window.eventTracker.track('click', {
        item: 'slider_price',
        section: 'filter_card',
        selection: `${value}`,
      })
    }
    this.props.updateFilterCriteriaWithChanges({ maxPrice: value })
    this.setState({ isSliding: false })
  }

  render() {
    const {
      priceBoundaries,
      defaultPriceRange,
      step,
    } = PRICE_FILTER_SLIDER_DEFAULTS

    const { theme } = this.props
    return (
      <PriceFilterCard
        data-tid="inline-price-filter-card"
        data-tag_section="filter_card"
        className={classnames(
          theme.InlineCard,
          this.state.isSliding && theme['InlinePriceFilterCard-sliding']
        )}
        title="Filter by price"
        description="Set a maximum price for your search."
        applyButton={{
          onClick: this.handleSubmit,
          'data-tag_item': 'update_results_button',
          'data-tag_selection': 'price',
        }}
        priceSlider={{
          'data-tid': 'range-slider',
          formatLabel: val => `$${val}`,
          value: priceBoundaries.value,
          minValue: defaultPriceRange.min,
          maxValue: defaultPriceRange.max,
          minLabel: defaultPriceRange.min,
          maxLabel: defaultPriceRange.max,
          step,
          onChangeStart: () => this.setState({ isSliding: true }),
        }}
        onChange={this.handlePriceChange}
      />
    )
  }
}
