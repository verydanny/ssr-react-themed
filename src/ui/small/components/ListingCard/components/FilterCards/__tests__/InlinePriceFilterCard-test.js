import React from 'react'
import { shallow } from 'enzyme'
import { PriceFilterCard } from '@rentpath/react-ui-core'
import InlinePriceFilterCard from '../InlinePriceFilterCard'

const defaultProps = {
  pushState: () => {},
  updateFilterCriteriaWithChanges: () => {},
}

const factory = props =>
  <InlinePriceFilterCard.WrappedComponent {...defaultProps} {...props} />

describe('small/ListingCard/FilterCards/components/InlinePriceFilterCard', () => {
  it('renders PriceFilterCard', () => {
    const wrapper = shallow(factory())
    expect(wrapper.find(PriceFilterCard)).toHaveLength(1)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
