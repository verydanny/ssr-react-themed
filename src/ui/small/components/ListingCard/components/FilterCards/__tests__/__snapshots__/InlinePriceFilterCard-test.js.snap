// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`small/ListingCard/FilterCards/components/InlinePriceFilterCard renders PriceFilterCard 1`] = `

<div class
     data-tid="inline-price-filter-card"
     data-tag_section="filter_card"
>
  <div data-tid="card-body">
    <div>
      <h2 class
          data-tid="title"
      >
        Filter by price
      </h2>
      <div data-tid="text"
           class
      >
        Set a maximum price for your search.
      </div>
    </div>
    <div>
      <div class>
        <div aria-disabled="false">
          <span>
            <span>
              $500
            </span>
          </span>
          <div>
            <div style="left:0%;width:0%">
            </div>
            <span style="position:absolute;left:0%">
              <span>
                <span>
                  $500
                </span>
              </span>
              <div aria-valuemax="5000"
                   aria-valuemin="500"
                   aria-valuenow="500"
                   draggable="false"
                   role="slider"
                   tabindex="0"
              >
              </div>
            </span>
          </div>
          <span>
            <span>
              $5000
            </span>
          </span>
        </div>
      </div>
      <div>
        <button data-tid="apply-button"
                data-tag_item="update_results_button"
                data-tag_selection="price"
                class
        >
          Apply
        </button>
      </div>
    </div>
  </div>
</div>

`;
